File: merge-intervals/merge-intervals.cs
public class Solution {
    public int[][] Merge(int[][] intervals) {
        List<int[]> lst = new List<int[]>();
        intervals = intervals.OrderBy(k => k[0]).ToArray();
        for(int i = 0; i < intervals.Length; i++) {
            if(lst.Count == 0 || lst[lst.Count - 1][1] < intervals[i][0]) {
                lst.Add(intervals[i]);
            }
            else {
                lst[lst.Count - 1][1] = Math.Max(lst[lst.Count - 1][1], intervals[i][1]);
            }
        }
        return lst.ToArray();
    }
}

File: merge-intervals/merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> a[0] - b[0]);
        List<int[]> lst = new ArrayList<>();
        lst.add(intervals[0]);
        for(int i = 1; i < intervals.length; i++) {
            int[] lastIv = lst.get(lst.size() - 1);
            if(intervals[i][0] <= lastIv[1]) {
                lastIv[1] = Math.max(lastIv[1], intervals[i][1]);
            }
            else {
                lst.add(intervals[i]);
            }
        }
        return lst.toArray(new int[lst.size()][]);
    }
}